cumulus_library/base_table_builder.py

```diff
@@ -102,19 +103,29 @@ def post_execution(
         """Hook for any additional actions to run after execute_queries"""
         pass
 
-    def comment_queries(self):
+    def comment_queries(self, doc_str=None):
         """Convenience method for annotating outputs of template generators to disk"""
         commented_queries = ["-- noqa: disable=all"]
+        if doc_str:
+            commented_queries.append(doc_str)
+            commented_queries.append(
+                "\n-- ###########################################################\n"
+            )
         for query in self.queries:
             commented_queries.append(query)
             commented_queries.append(
-                "\n-- ###########################################################"
+                "\n-- ###########################################################\n"
             )
         commented_queries.pop()
         self.queries = commented_queries
 
-    def write_queries(self, filename: str = "output.sql"):
+    def write_queries(
+        self, filename: str = "output.sql", dir_path: pathlib.Path = None
+    ):
         """writes all queries constructed by prepare_queries to disk"""
+        if dir_path:
+            dir_path.mkdir(exist_ok=True)
+            filename = dir_path / filename
```

Comment:
nit: out of curiousity, does `path /= filename` work? Feels like I wouldn't want that to work. Just wondering if Python lets you handle the operator `/` but disallow `/=`.

Reply:
Looks like it doesn't.